repos/system_upgrade/el7toel8/actors/migratebrltty/libraries/library.py

```diff
@@ -0,0 +1,24 @@
+import re
+import shutil
+
+BackupSuffix = '.bak'
+
+def migrate_file(fn, migrate_bt, migrate_espeak):
+    # nothing to migrate
+    if not migrate_bt and not migrate_espeak:
+        return
+
+    # make backup
+    shutil.copy2(fn, fn + BackupSuffix)
+
+    regex_bt = re.compile(r'\b(?:(?:bth)|(?:bluez))((?:[:\-][0-9a-fA-F]{2}){6})\b')
+    regex_espeak = re.compile(r'^(\s*speech-driver\s+)es\b')
+
+    with open(fn, 'w') as file_out:
+        with open(fn + BackupSuffix) as file_in:
+            for line in file_in:
+                if migrate_bt and regex_bt.search(line):
+                    line = regex_bt.sub(r'bluetooth\1', line)
+                if migrate_espeak and regex_espeak.search(line):
```

Comment:
Open question: Can it be elif? In other words - can 2 regex match the same line?

Reply:
I think here, it can be used as an optimization.

Fixed.